lib/ansible/modules/network/panos/panos_security_policy.py

```diff
@@ -0,0 +1,516 @@
+        default: "allow"
+    group_profile:
+        description: >
+            security profile group that is already defined in the system. This property supersedes antivirus,
+            vulnerability, spyware, url_filtering, file_blocking, data_filtering, and wildfire_analysis properties.
+        required: false
+        default: None
+    antivirus:
+        description:
+            - Name of the already defined antivirus profile.
+        required: false
+        default: None
+    vulnerability:
+        description:
+            - Name of the already defined vulnerability profile.
+        required: false
+        default: None
+    spyware:
+        description:
+            - Name of the already defined spyware profile.
+        required: false
+        default: None
+    url_filtering:
+        description:
+            - Name of the already defined url_filtering profile.
+        required: false
+        default: None
+    file_blocking:
+        description:
+            - Name of the already defined file_blocking profile.
+        required: false
+        default: None
+    data_filtering:
+        description:
+            - Name of the already defined data_filtering profile.
+        required: false
+        default: None
+    wildfire_analysis:
+        description:
+            - Name of the already defined wildfire_analysis profile.
+        required: false
+        default: None
+    devicegroup:
+        description: >
+            Device groups are used for the Panorama interaction with Firewall(s). The group must exists on Panorama.
+            If device group is not define we assume that we are contacting Firewall.
+        required: false
+        default: None
+    commit:
+        description:
+            - commit if changed
+        required: false
+        default: true
+'''
+
+EXAMPLES = '''
+# permit ssh to 1.1.1.1
+- panos_security_policy:
+    ip_address: '10.5.172.91'
+    username: 'admin'
+    password: 'paloalto'
+    rule_name: 'SSH permit'
+    description: 'SSH rule test'
+    from_zone: ['public']
+    to_zone: ['private']
+    source: ['any']
+    source_user: ['any']
+    destination: ['1.1.1.1']
+    category: ['any']
+    application: ['ssh']
+    service: ['application-default']
+    hip_profiles: ['any']
+    action: 'allow'
+    commit: false
+
+# Allow HTTP multimedia only from CDNs
+- panos_security_policy:
+    ip_address: '10.5.172.91'
+    username: 'admin'
+    password: 'paloalto'
+    rule_name: 'HTTP Multimedia'
+    description: 'Allow HTTP multimedia only to host at 1.1.1.1'
+    from_zone: ['public']
+    to_zone: ['private']
+    source: ['any']
+    source_user: ['any']
+    destination: ['1.1.1.1']
+    category: ['content-delivery-networks']
+    application: ['http-video', 'http-audio']
+    service: ['service-http', 'service-https']
+    hip_profiles: ['any']
+    action: 'allow'
+    commit: false
+
+# more complex fictitious rule that uses profiles
```

Comment:
Examples should specify a `- name:` rather than a comment, please see http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#examples-block for details.